common/src/main/java/io/quarkus/ts/openshift/common/util/ReadinessUtil.java

```diff
@@ -9,21 +9,34 @@
 /**
  * Unifies Fabric8 Kubernetes Client's {@link Readiness} and {@link OpenShiftReadiness}.
  * No other class should refer to them directly; using {@code ReadinessUtil} is preferred.
+ * Note: this proxy is strongly couple to `io.fabric8.kubernetes.client.internal.readiness`.
+ * We are making a virtual invocation with version 5.1.0 and a static invocation with version < 5.0.X, something
+ * that is `quite hacky`. Please consider refactor this proxy, once `Readiness isReadinessApplicable` method become public.
  */
-public class ReadinessUtil {
-    public static boolean isReadinessApplicable(Class<? extends HasMetadata> resourceClass) {
-        // unfortunately `OpenShiftReadiness` doesn't have a `isReadinessApplicable` method
-        return Readiness.isReadinessApplicable(resourceClass)
-                || DeploymentConfig.class.isAssignableFrom(resourceClass)
-                ;
+public enum ReadinessUtil {
+
+    INSTANCE;
+
+    private final Readiness readiness;
+
+    ReadinessUtil(){
+        readiness = new Readiness();
+    }
+
+    public boolean isReadinessApplicable(HasMetadata resource) {
+        try{
+            return readiness.isReady(resource)|| DeploymentConfig.class.isAssignableFrom(resource.getClass());
```

Comment:
Ah, I didn't notice this before (and if I did, I wouldn't approve). Calling `readiness.isReady(resource)` just for `isReadinessApplicable` is not a good idea. This method isn't supposed to talk to the Kubernetes API server, it's supposed to be a quick check whether the readiness concept even makes sense for this kind of resource.

Please just abandon the entire idea that you can have a single class that will work with both Fabric8 Kubernetes Client 5.0.x and 5.1.x. They made a breaking change, why pretend they didn't.

Reply:
Ah one more thing. If I read this correctly, then this method can return `false` for resources where readiness does make sense (and hence the method should return `true`), but are not ready at that particular point in time. This is just bad.